esx_service/cli/vmdkops_admin.py

```diff
@@ -1301,16 +1305,38 @@ def config_rm(args):
     # This asks for double confirmation, and removes the local link or DB (if any)
     # NEVER deletes the shared database - instead prints help
 
-    if not args.local:
+    if not args.local and not args.unlink:
         return err_out("""
         Shared DB removal is not supported. For removing  local configuration, use --local flag.
-        For removing shared DB,  run 'vmdkops_admin config rm --local' on ESX hosts using this DB,
+        For removing shared DB,  run 'vmdkops_admin config rm --unlink' on ESX hosts using this DB,
         and manually remove the {} file from shared storage.
         """.format(auth_data.CONFIG_DB_NAME))
 
+    # Check the existing config mode
+    with auth_data.AuthorizationDataManager() as auth:
+        try:
+            auth.connect()
+            info = auth.get_info()
+            mode = auth.mode # for usage outside of the 'with'
+        except auth_data.DbAccessError as ex:
+            return err_out(str(ex))
+
     if not args.confirm:
         return err_out("Warning: For extra safety, removal operation requires '--confirm' flag.")
 
+    if mode == auth_data.DBMode.NotConfigured:
+        pass
+    elif mode == auth_data.DBMode.MultiNode:
+        if args.local:
+            return err_out("'rm --local' is not supported when " + DB_REF + "is in MultiNode mode."
+                        "Use 'rm --unlink' to remove the local link to shared DB.")
+    elif mode == auth_data.DBMode.SingleNode:
+        if args.unlink:
+            return err_out("'rm --unlink' is not supported when " + DB_REF + " is in SingleNode mode."
+                        " Use 'rm --local' to remove the local link or local DB.")
```

Comment:
Why "local link or local DB"? Do we also create a symlink to local DB? If yes, we should say "remove local link and local DB" or simply "remove local DB configuration"